### ISSUE TITLE
Align MD009 option "strict" with markdownlint behaviour

### ISSUE THREAD

**Issue by palto42** [rvben/rumdl#593]
### Problem or motivation

Although rumdl claims to be fully compliant with markdownlint, the MD009 option "strict" currently behaves differently.

In markdownlint, the strict option is only triggered for unnecessary br-spaces, while in rumdl it is always triggered.

Example markdown with 2 spaces for extra line break, which is not unnecessary and passes markdownlint even with MD009 strict=true, but fails with same setting in rumdl.

```markdown
> Note:  
> This is in a new line due to 2 spaces behind "Note:  ".
```

### Proposed solution

Align with behavior in markdownlint.

### Alternatives considered

Update documentation with list of differences between rumdl and markdownlint.

### Feature category

New lint rule

### Example markdown

```markdown

```

### Contribution

- [ ] I would be willing to submit a PR to implement this feature

**Comment by rvben** [comment]
Fixed in v0.1.88. Thanks for the report, @palto42!

In strict mode, trailing br-spaces are now permitted when they produce a meaningful line break — specifically, when the current and next lines are both in paragraph context. Hard line breaks inside paragraphs are intentional and should not be flagged as trailing whitespace.
